### ISSUE TITLE
Weave-Net dashboards only show "No Data"

### ISSUE THREAD

**Issue by thkukuk** [coreos/kube-prometheus#446]
**What happened?**
The new Weave-Net and Weave-Net (Cluster) dashboards show both "No Data".

**Did you expect to see some different?**
I expect to see the data.

**How to reproduce it (as minimally and precisely as possible)**:
Deploy weave with the official yaml manifests as provided with the weave sources.
Deploy the resulting manifests of examples/weave-net-example.jsonnet from current kube-prometheus git.

**Environment**

* Kubernetes version information: 1.17.2
* Kubernetes cluster kind: kubeadm

**Anything else we need to know?**:

The problem seems to be a mismatch of labels. weave.yaml from weave-2.6.1 uses as label "name=weave-net", while e.g. prometheus-serviceWeaveNet.yaml uses for the selector "k8s-app=weave-net". So the service will not match the weave deployment.
I tried to fix the obvious places, but afterwards it's still not working.

I can query the weave metrics from the weave pods manually and they contain usefull data, so that part is working.

**Comment by lilic** [comment]
cc @alok87 who contributed this dashboard

**Comment by alok87** [comment]
@thkukuk thanks for reporting, let me check.

@lilic Please assign this issue to me.

**Comment by alok87** [comment]
@thkukuk have you created the weave service? The manifest would have been created with `prometheus-serviceWeaveNet.yaml` name.

The label is for this service.
```
apiVersion: v1
kind: Service
metadata:
  labels:
    k8s-app: weave-net
  name: weave-net
  namespace: kube-system
```

**Comment by thkukuk** [comment]
I applied prometheus-serviceWeaveNet.yaml but this service has as last lines:
```
  selector:
    k8s-app: weave-net
```
But weave.yaml uses as label:
```
      labels:
        name: weave-net
```
As result, the service will never match the weave.yaml deployment. You can check yourself: nslookup weave-net.kube-system is not resolveable with the original prometheus-serviceWeaveNet.yaml, but it is if I change the selector to "name: weave-net".
After this, I can query weave-net.kube-system:6782/metrics and get correct output.

But this does not seem to be enough. prometheus/grafana still report no data. So I assume prometheus-serviceMonitorWeaveNet.yaml contains a similar error. But I don't have any experience with ServiceMonitor.

**Comment by alok87** [comment]
@thkukuk I have verified the fix #448 in my cluster. It works!

**Comment by thkukuk** [comment]
#448 works for me, too.
